### PR TITLE
Extended search

### PR DESCRIPTION
**Pull request by hboutemy**
Extended search in library.$name.path to do the same as in META-INF/native:
this will enable scenario where a distribution extracts native libraries to avoid temp file creation if the lib auto-extracts on each run
see https://issues.apache.org/jira/browse/MNG-6186 for a use case in Maven

### CODE REVIEW

File: hawtjni-runtime/src/main/java/org/fusesource/hawtjni/runtime/Library.java [hunk 1]
```diff
@@ -20,27 +20,36 @@
...
+ *   <li> "<code></code>"
+ *   </ol>
+ *   for 2 namings of the library:
  *   <ol>
  *   <li> as "<code>${name}-${version}</code>" library name if the version can be determined.
  *   <li> as "<code>${name}</code>" library name
  *   </ol>
  * <li> system library path: This is where the JVM looks for JNI libraries by default.
  *   <ol>
+ *   <li> as "<code>${name}${bit-model}-${version}</code>" library name if the version can be determined.
```
Comment (michael-o) [inline hawtjni-runtime/src/main/java/org/fusesource/hawtjni/runtime/Library.java]: Why did you add that? System library path (`LD_LIBRARY_PATH`, `rpath`) will unlikely contain the bit model. It looks unnecessary to me.

File: hawtjni-runtime/src/main/java/org/fusesource/hawtjni/runtime/Library.java [hunk 2]
```diff
@@ -20,27 +20,36 @@
...
  *   <li> as "<code>${name}-${version}</code>" library name if the version can be determined.
  *   <li> as "<code>${name}</code>" library name
  *   </ol>
  * <li> classpath path: If the JNI library can be found on the classpath, it will get extracted
  * and then loaded. This way you can embed your JNI libraries into your packaged JAR files.
  * They are looked up as resources in this order:
  *   <ol>
- *   <li> "<code>META-INF/native/${platform}/${arch}/${library}</code>": Store your library here if you want to embed
+ *   <li> "<code>META-INF/native/${platform}/${arch}/${library[-version]}</code>": Store your library here if you want to embed
```
Comment (michael-o) [inline hawtjni-runtime/src/main/java/org/fusesource/hawtjni/runtime/Library.java]: How is this version necessary for our usecase? Especially because only one version of the native library should exist once in the classpath OR the JAR from central contains its version anyway.

File: hawtjni-runtime/src/main/java/org/fusesource/hawtjni/runtime/Library.java (hunk 1 above)
Comment (hboutemy) [inline hawtjni-runtime/src/main/java/org/fusesource/hawtjni/runtime/Library.java]: why did I add to documentation what was done in code?

File: hawtjni-runtime/src/main/java/org/fusesource/hawtjni/runtime/Library.java (hunk 2 above)
Comment (hboutemy) [inline hawtjni-runtime/src/main/java/org/fusesource/hawtjni/runtime/Library.java]: once again, I'm not judging anything: I'm just keeping existing features, even if I never used it myself

**Comment by michael-o** [comment]
Looks good to me after review.

**Comment by gnodet** [comment]
Could you rebase your PR please ?

**Comment by gnodet** [comment]
Manually merged.
